fail2ban/server/filter.py

```diff
@@ -855,8 +855,9 @@ def dnsToIp(dns):
 		""" Convert a DNS into an IP address using the Python socket module.
 			Thanks to Kevin Drapel.
 		"""
+		# retrieve ip (todo: use AF_INET6 for IPv6)
 		try:
-			return set(socket.gethostbyname_ex(dns)[2])
+			return set([i[4][0] for i in socket.getaddrinfo(dns, None, socket.AF_INET, 0, socket.IPPROTO_TCP)])
```

Comment:
i.e. wouldn't we might miss some entries?

but why fail2ban was getting stuck? socket.gethostbyname_ex was getting stuck for me it just pukes:
```
In [2]: socket.gethostbyname_ex('unassigned.infomir.com.ua')
---------------------------------------------------------------------------
herror                                    Traceback (most recent call last)
<ipython-input-2-8e5a8f862f26> in <module>()
----> 1 socket.gethostbyname_ex('unassigned.infomir.com.ua')

herror: [Errno 1] Unknown host
```
so should have returned an empty list and nothing indeed would have been banned (but shouldn't get stuck).  Now it would return 1000s of IPs and ban ALL of them... not sure which behaviour is better (and we do recommend to not rely on names but rather log IPs), but the question -- what /why was "stuck"??